java/compaction-status-store/src/main/java/sleeper/compaction/status/task/DynamoDBCompactionTaskStatusStore.java

```diff
@@ -15,11 +15,85 @@
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.PutItemResult;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryResult;
+import com.amazonaws.services.dynamodbv2.model.ReturnConsumedCapacity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sleeper.compaction.status.CompactionStatusStoreException;
+import sleeper.compaction.task.CompactionTaskStatus;
+import sleeper.compaction.task.CompactionTaskStatusStore;
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.UserDefinedInstanceProperty;
+
+import java.util.Map;
+
+import static sleeper.compaction.status.DynamoDBAttributes.createStringAttribute;
 import static sleeper.compaction.status.DynamoDBUtils.instanceTableName;
+import static sleeper.compaction.status.task.DynamoDBCompactionTaskStatusFormat.TASK_ID;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.ID;
+
+public class DynamoDBCompactionTaskStatusStore implements CompactionTaskStatusStore {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamoDBCompactionTaskStatusStore.class);
+    private final AmazonDynamoDB dynamoDB;
+    private final String statusTableName;
+    private Long timeToLive;
+
+    private DynamoDBCompactionTaskStatusStore(AmazonDynamoDB dynamoDB, InstanceProperties properties) {
+        this.dynamoDB = dynamoDB;
+        this.statusTableName = taskStatusTableName(properties.get(ID));
+        this.timeToLive = properties.getLong(UserDefinedInstanceProperty.COMPACTION_JOB_STATUS_TTL_IN_SECONDS) * 1000;
+    }
+
+    @Override
+    public void taskStarted(CompactionTaskStatus taskStatus) {
+        try {
+            PutItemResult result = putItem(DynamoDBCompactionTaskStatusFormat.createTaskStartedRecord(taskStatus, taskStatus.getStartedStatus().getStartTime(), timeToLive));
+            LOGGER.debug("Put created event for job {} to table {}, capacity consumed = {}",
+                    taskStatus.getTaskId(), statusTableName, result.getConsumedCapacity().getCapacityUnits());
+        } catch (RuntimeException e) {
+            throw new CompactionStatusStoreException("Failed putItem in taskCreated", e);
+        }
+    }
+
+    @Override
+    public void taskFinished(CompactionTaskStatus taskStatus) {
+        try {
+            PutItemResult result = putItem(DynamoDBCompactionTaskStatusFormat.createTaskFinishedRecord(taskStatus, taskStatus.getFinishedStatus().getFinishTime(), timeToLive));
+            LOGGER.debug("Put created event for job {} to table {}, capacity consumed = {}",
+                    taskStatus.getTaskId(), statusTableName, result.getConsumedCapacity().getCapacityUnits());
+        } catch (RuntimeException e) {
+            throw new CompactionStatusStoreException("Failed putItem in taskCreated", e);
```

Comment:
The exception and log messages here should be different from the other method.

They should both refer to tasks rather than jobs.

Reply:
They still say created and taskCreated instead of started and finished.